code/modules/admin/verbs/request_music.dm

```diff
@@ -0,0 +1,33 @@
+/mob/verb/request_music()
+	set category = "OOC"
+	set name = "Request Music"
+
+	if(GLOB.say_disabled) //This is here to try to identify lag problems
+		to_chat(usr, span_danger("Speech is currently admin-disabled."), confidential = TRUE)
+		return
+
+	var/request_url = tgui_input_text(usr.client, "Please Input a URL", "Music Request", "")
+	if(!request_url)
+		return
+
+	//regex filter
+	if(!findtext(request_url, "youtube.com/watch?v=") && !findtext(request_url, "youtu.be/") && !findtext(request_url, "soundcloud.com/"))
+		to_chat(usr, span_danger("Invalid URL. Please use a YouTube URL, or Soundcloud URL"), confidential = TRUE)
```

Comment:
This must be a config.

You are not allowed to hardcode a list of supported sites.